packages/headless/src/features/analytics/search-analytics-actions-loader.ts

```diff
@@ -391,6 +398,71 @@ export interface SearchAnalyticsActionCreators {
     void,
     AsyncThunkAnalyticsOptions<StateNeededByAnalyticsProvider>
   >;
+
+  /**
+   * The event to log when a smart snipped is collapsed.
+   *
+   * @returns A dispatchable action.
+   */
+  logCollapseSmartSnippet(): AsyncThunkAction<
+    {
+      analyticsType: AnalyticsType.Custom;
+    },
+    void,
+    AsyncThunkAnalyticsOptions<StateNeededByAnalyticsProvider>
+  >;
+
+  /**
+   * The event to log when a smart snipped is expanded.
+   *
+   * @returns A dispatchable action.
+   */
+  logExpandSmartSnippet(): AsyncThunkAction<
+    {
+      analyticsType: AnalyticsType.Custom;
+    },
+    void,
+    AsyncThunkAnalyticsOptions<StateNeededByAnalyticsProvider>
+  >;
+
+  /**
+   * The event to log when a user provides detailed feedback for a given smart snippet answer.
+   *
+   * @returns A dispatchable action.
+   */
+  logSmartSnippetExplanationFeedback(): AsyncThunkAction<
+    {
+      analyticsType: AnalyticsType.Custom;
+    },
+    void,
+    AsyncThunkAnalyticsOptions<StateNeededByAnalyticsProvider>
+  >;
+
+  /**
+   * The event to log when a user provides negative feedback for a given smart snippet answer.
+   *
+   * @returns A dispatchable action.
+   */
+  logSmartSnippetNegativeFeedback(): AsyncThunkAction<
+    {
+      analyticsType: AnalyticsType.Custom;
+    },
+    void,
+    AsyncThunkAnalyticsOptions<StateNeededByAnalyticsProvider>
+  >;
+
+  /**
+   * The event to log when a user provides positive feedback for a given smart snippet answer.
+   *
+   * @returns A dispatchable action.
+   */
+  logSmartSnippetPositiveFeedback(): AsyncThunkAction<
```

Comment:
If we plan on not relying on state in the future, should we make the new actions accept their payload upfront? Just a thought ...

Reply:
Those won't require any additional metadata (so, no reading from the state), apart from the standard search ID.

So, there's not much gain to do it here "just for these ones".

Reply:
Alright, the more I think about the general problem, maybe a one size fits all solution is not best. Perhaps some actions could read from state to keep an easier api, while certain ones can be adjusted to need their payload, if there is a high chance of order-issues.